src/vm/i386/asmhelpers.S

```diff
@@ -1138,3 +1138,88 @@ NESTED_ENTRY ResolveWorkerAsmStub, _TEXT, NoHandler
     // Now jump to the target
     jmp     eax             // continue on into the method
 NESTED_END ResolveWorkerAsmStub, _TEXT
+
+NESTED_ENTRY ResolveWorkerChainLookupAsmStub, _TEXT, NoHandler
+
+#define CALL_STUB_CACHE_INITIAL_SUCCESS_COUNT   0x100
+
+#define e_mt_size               4
+#define e_token_size            4
+#define e_target_size           4
+#define e_indirect_addr_size    4
+#define e_caller_ret_addr_size  4
+
+// this is the part of the stack that is present as we enter this function:
+#define e_token                 0
+#define e_indirect_addr         (e_token + e_token_size)
+#define e_caller_ret_addr       (e_indirect_addr + e_indirect_addr_size)
+#define e_ret_esp               (e_caller_ret_addr + e_caller_ret_addr_size)
+
+#define e_offset_mt             0
```

Comment:
Could you please put all of these offsets into the asmconstants.h together with their verification like we do for other similar offsets and sizes and name them using the same convention we use for the others (like `OFFSETOF__ResolveCacheElem_target` etc)? 
Also, I would prefer not using the `e_` prefix on the stack offsets, since they are not enums like in the Windows version of this code. Maybe you could use a similar naming convention like we use in the umthunkstub.S for stack offsets.